test/spec/services/basket.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+describe('Service: Basket', function () {
+
+  // load the service's module
+  beforeEach(module('tmaClientApp'));
+
+  // instantiate service
+  var mockBasket;
+
+  beforeEach(inject(function(Basket) {
+    mockBasket = Basket;
+  }));
+
+  it('should initially be empty', function(){
+    expect(mockBasket.count()).toBe(0);
+  });
+
+  describe('Basket.get()', function() {
+    it('should have a get function', function() {
+      expect(angular.isFunction(mockBasket.get)).toBe(true);
+    });
+  });
+
+  describe('Basket.add()', function() {
+    var item1 = { id: 2, name: 'Aloo Matar', price: 59.0 };
+
+    it('should have a function to add new items to basket', function() {
+      expect(angular.isFunction(mockBasket.add)).toBe(true);
+    });
+
+    it('should add items to the basket', function() {
+      expect(mockBasket.get()).toEqual({});
+      mockBasket.add(item1);
+      expect(mockBasket.count()).toBe(1);
+    });
+
+    it('should update item quantity if item already in basket', function() {
+      expect(mockBasket.get()).toEqual({});
+      mockBasket.add(item1);
+      mockBasket.add(item1);
+      expect(mockBasket.count()).toBe(1);
+
+      var basket = mockBasket.get();
+      expect(basket[item1.name].quantity).toBe(2);
```

Comment:
in my opinion, it might be better if the `Basket` service does not expose its implementation (i.e. the app does not know that internally it uses a plain object to store the details). So something like:
```javascript
mockBasket.quantityOf(item1.name)
```